components/PortalStatistics.lua

```diff
@@ -0,0 +1,1225 @@
+		:css('text-align', 'center')
+		:wikitext('Tournament Type')
+		:node(StatisticsPortal._getPieChartData(
+			args, 'type', 'Mixed', TYPES
+		))
+	)
+
+	if args.multiGame then
+		wrapper:node(mw.html.create('div')
+			:addClass('template-box')
+			:css('padding-right','5em')
+			:css('width', args.width)
+			:css('font-size', '85%')
+			:css('text-align', 'center')
+			:wikitext('Game Breakdown')
+			:node(StatisticsPortal._getPieChartData(
+				args, 'game', 'Other', StatisticsPortal._splitOrDefault(args.customGames, GAMES)
+			))
+		)
+	end
+
+	if args.multiMode then
+		wrapper:node(mw.html.create('div')
+			:addClass('template-box')
+			:css('padding-right','5em')
+			:css('width', args.width)
+			:css('font-size', '85%')
+			:css('text-align', 'center')
+			:wikitext('Mode Breakdown')
+			:node(StatisticsPortal._getPieChartData(
+				args, 'mode', 'Other', StatisticsPortal._splitOrDefault(args.customModes, {'Team'})
+			))
+		)
+	end
+
+	if Logic.readBool(args.hideKey) then
+		return wrapper
+	end
+
+	if Logic.readBool(args.detailedKey) then
+		wrapper:node(mw.html.create('div')
+			:addClass('template-box')
+			:node(StatisticsPortal.prizepoolBreakdown(args))
+		)
+		return wrapper
+	end
+
+	local conditions = StatisticsPortal._returnBaseConditions()
+
+	if args.year then
+		conditions:add{ConditionTree(BooleanOperator.all):add{
+				ConditionNode(ColumnName('sortdate_year'), Comparator.eq, args.year),
+			},
+		}
+	else
+		conditions:add{ConditionTree(BooleanOperator.all):add{
+				ConditionNode(ColumnName('sortdate'), Comparator.lt, DATE),
+			},
+		}
+	end
+
+	if args.game then
+		conditions:add{ConditionNode(ColumnName('game'), Comparator.eq, args.game)}
+	end
+
+	local data = mw.ext.LiquipediaDB.lpdb('tournament', {
+		query = 'sum::prizepool',
+		limit = 5000,
+		conditions = conditions:toString(),
+		order = 'sortdate desc',
+	})
+
+	local summaryTable = mw.html.create('table')
+		:addClass('wikitable')
+		:css('text-align','center')
+		:css('font-weight','bold')
+
+	summaryTable:tag('tr')
+		:tag('th'):wikitext('Total prize money awarded')
+
+	summaryTable:tag('tr')
+		:tag('td')
+		:wikitext(Currency.display('USD', data[1].sum_prizepool or 0, CURRENCY_FORMAT_OPTIONS))
+		:attr('data-sort-type', 'currency')
+		:css('font-weight','bold')
+
+	wrapper:node(mw.html.create('div')
+		:addClass('template-box')
+		:css('padding-right','1em')
+		:node(summaryTable)
+	)
+
+	return wrapper
+end
+
+
+function StatisticsPortal.earningsTable(args)
+	args = args or {}
+	args.limit = tonumber(args.limit) or 20
+	args.opponentType = args.opponentType or Opponent.team
+	args.displayShowMatches = Logic.readBool(args.displayShowMatches)
+	args.allowedPlacements = StatisticsPortal._splitOrDefault(args.allowedPlacements, DEFAULT_ALLOWED_PLACES)
+
+	local earningsFunction = function (a)
+		if String.isNotEmpty(args.year) and a.extradata then
+			return tonumber(a.extradata['earningsin'..args.year] or 0)
+		else
+			return tonumber(a.earnings or 0)
```

Comment:
```suggestion
			return tonumber(a.extradata['earningsin'..args.year]) or 0
		else
			return tonumber(a.earnings) or 0
```
in case the values you `tonumber` are set but not numeric this catches it
(better safe than sorry)